noir_stdlib/src/meta/op.nr

```diff
@@ -26,6 +26,20 @@ impl UnaryOp {
         // docs:end:is_dereference
         self.op == 3
     }
+
+    // docs:start:unary_quoted
+    pub fn quoted(self) -> Quoted {
+        // docs:end:unary_quoted
+        if self.is_minus() {
+            quote { - }
+        } else if self.is_not() {
+            quote { ! }
+        } else if self.is_mutable_reference() {
+            quote { &mut }
+        } else {
+            quote { * }
```

Comment:
```suggestion
        } else if self.is_dereference() {
            quote { * }
        } else {
            assert(false, "unhandled unary operator");
        }
```
This makes it more future-proof if we ever add unary operators

Reply:
Great! I forgot we had a way to kind of `panic!` at compile-time.

Reply:
Oh, it doesn't seem to work:

```
error: Expected type Quoted, found type ()
   ┌─ std/meta/op.nr:41:16
   │
41 │           } else {
   │ ╭────────────────'
42 │ │             assert(false, "unhandled unary operator")
43 │ │         }
   │ ╰─────────'
```

I guess assert's type is `()`? Do we have a bottom type (`!`) like in Rust?